tests/bluetooth/tester/src/gatt.c

```diff
@@ -1689,42 +1708,53 @@ static uint8_t notify_func(struct bt_conn *conn,
 static void discover_complete(struct bt_conn *conn,
 			      struct bt_gatt_discover_params *params)
 {
+	struct bt_gatt_subscribe_params *subscription;
 	uint8_t op, status;
 
+	subscription = find_subscription(discover_params.end_handle);
+	__ASSERT_NO_MSG(subscription);
+
 	/* If no value handle it means that chrc has not been found */
-	if (!subscribe_params.value_handle) {
+	if (!subscription->value_handle) {
 		status = BTP_STATUS_FAILED;
 		goto fail;
 	}
 
-	if (bt_gatt_subscribe(conn, &subscribe_params) < 0) {
+	if (bt_gatt_subscribe(conn, subscription) < 0) {
 		status = BTP_STATUS_FAILED;
 		goto fail;
 	}
 
 	status = BTP_STATUS_SUCCESS;
 fail:
-	op = subscribe_params.value == BT_GATT_CCC_NOTIFY ? GATT_CFG_NOTIFY :
-							    GATT_CFG_INDICATE;
+	op = subscription->value == BT_GATT_CCC_NOTIFY ? GATT_CFG_NOTIFY :
+							 GATT_CFG_INDICATE;
 
 	if (status == BTP_STATUS_FAILED) {
-		(void)memset(&subscribe_params, 0, sizeof(subscribe_params));
+		(void)memset(subscription, 0, sizeof(*subscription));
 	}
 
 	tester_rsp(BTP_SERVICE_ID_GATT, op, CONTROLLER_INDEX, status);
+
+	(void)memset(params, 0, sizeof(*params));
 }
 
 static uint8_t discover_func(struct bt_conn *conn,
 			     const struct bt_gatt_attr *attr,
 			     struct bt_gatt_discover_params *params)
 {
+	struct bt_gatt_subscribe_params *subscription;
+
 	if (!attr) {
 		discover_complete(conn, params);
 		return BT_GATT_ITER_STOP;
 	}
 
+	subscription = find_subscription(discover_params.end_handle);
+	__ASSERT_NO_MSG(subscription);
+
 	/* Characteristic Value Handle is the next handle beyond declaration */
-	subscribe_params.value_handle = attr->handle + 1;
+	subscription->value_handle = attr->handle + 1;
```

Comment:
You can also consider using the CONFIG_BT_GATT_AUTO_DISCOVER_CCC feature to determine the CCC handle